pkg/metadata/finalizers.go

```diff
@@ -44,32 +45,18 @@ func HasResourceGraphDefinitionFinalizer(obj metav1.Object) bool {
 
 // SetInstanceFinalizerUnstructured adds an instance-specific finalizer to an unstructured object.
 func SetInstanceFinalizerUnstructured(obj *unstructured.Unstructured) error {
-	finalizers, found, err := unstructured.NestedStringSlice(obj.Object, "metadata", "finalizers")
-	if err != nil {
-		return fmt.Errorf("error getting finalizers: %w", err)
-	}
-
-	if !found || !containsString(finalizers, kroFinalizer) {
+	finalizers := obj.GetFinalizers()
```

Comment:
this finalizer code looks unrelated to the PR, better to pull it to a seperate PR for better visibility.

Reply:
https://github.com/kro-run/kro/pull/625